packages/backend/src/routes/apiProxy.ts

```diff
@@ -67,7 +67,22 @@ export const proxyRequestHandler = async (
       // we must apply the template values to the body as well
       try {
         const bodyText = await req.text()
-        templateBody = applyTemplateValues(bodyText, requestCookies)
+        const contentType = req.headers.get('content-type')?.toLowerCase()
+
+        if (contentType?.includes('application/x-www-form-urlencoded')) {
+          // Parse form data, apply templates to values, then re-encode
+          templateBody = new URLSearchParams(
+            Object.fromEntries(
+              [...new URLSearchParams(bodyText)].map(([key, value]) => [
```

Comment:
More of an opinion than correctness, but you could change this to

```suggestion
              new URLSearchParams(bodyText).entries().map(([key, value]) => [
```